### ISSUE TITLE
pyam.concat / pyam.append overwrites and not updates year parameter 

### ISSUE THREAD

**Issue by l-welder** [IAMconsortium/pyam#485]
the concat command has issues:

```
IAMC_IDX = ['model', 'scenario', 'region', 'variable', 'unit']

TEST_YEARS_PROJ = [2005, 2010]
TEST_YEARS_HIST = [2000]

proj = pd.DataFrame([
    ['model_a', 'scen_a', 'reg', 'PE', 'EJ/yr', 1, 6.],
],
    columns=IAMC_IDX + TEST_YEARS_PROJ,
)
proj = pyam.IamDataFrame(proj)

hist = pd.DataFrame([
    ['source_b', 'scen_b', 'reg', 'PE', 'EJ/yr', 0.5],
],
    columns=IAMC_IDX + TEST_YEARS_HIST,
)
hist = pyam.IamDataFrame(hist)

concat = pyam.concat([hist, proj])

print(concat)
print(concat.year)
```

gives me

```
 <class 'pyam.core.IamDataFrame'>
Index dimensions:
 * model    : model_a, source_b (2)
 * scenario : scen_a, scen_b (2)
Timeseries data coordinates:
   region   : reg (1)
   variable : PE (1)
   unit     : EJ/yr (1)
   year     : 2000, 2005, 2010 (3)
Meta indicators:
   exclude (bool) False (1)

 [2000]
```

(the latter should be [2000, 2005 and 2010]). The issue only arises for .year

**Comment by danielhuppmann** [comment]
Thanks @l-welder for reporting this issue - I also noticed this last week and fixed it (silently) in #481, by pulling out the attribute-definition into an own function `_set_attributes()` (which is called after `append()` and other functions that implicitly modify the `_data` attribute).

Do you have a minute to take a look at #481?

**Comment by l-welder** [comment]
I am a bit short of time right now for a deeper look into the PR, but I pulled your branch and with that the issue is fixed (as you already suggested) (=
